lib/gateway.js

```diff
@@ -281,6 +288,47 @@ async function buildGateway (gatewayOpts, app) {
     schema,
     serviceMap,
     entityResolversFactory: factory,
+    pollingInterval: gatewayOpts.pollingInterval,
+    async refresh () {
+      if (this._serviceSDLs === undefined) {
+        this._serviceSDLs = serviceSDLs.join('')
+      }
+
+      await allSettled(
+        Object.values(serviceMap).map((service) =>
+          service.refresh().catch((err) => {
+            if (service.mandatory) {
+              throw Object.assign(err, service)
```

Comment:
I'd like to hear @mcollina opinion on that as well, as I'd like to hear yours @iviaks:

I think it might be beneficial if we somehow had the ability to log (or somehow react on) the situation when a service (that's not flagged as `mandatory`) isn't reachable. Maybe just to add it to my metrics, so that I could identify patterns or to my monitoring in order to identify potential problems.

Reply:
It's a good thing.

We can use `errorHandler` function for getting everything and the developer will be able to use it as he wants.
For example, Developer wants to log all mandatory errors, so `errorHandler` should be:

```js
const errorHandler = (error, service) => {
  if (service.mandatory) {
    logger.warn(error)
  }
}
```

Changes according to my words.

```diff
diff --git a/lib/gateway.js b/lib/gateway.js
index b9185a0..cdeac26 100644
--- a/lib/gateway.js
+++ b/lib/gateway.js
@@ -297,16 +297,10 @@ async function buildGateway (gatewayOpts, app) {
       await allSettled(
         Object.values(serviceMap).map((service) =>
           service.refresh().catch((err) => {
-            if (service.mandatory) {
-              throw Object.assign(err, service)
-            }
+            errorHandler(err, service)
           })
         )
-      ).then((result) => {
-        result
-          .filter(({ reason }) => reason !== undefined)
-          .forEach(({ reason }) => errorHandler(reason, reason.service))
-      })
+      )
 
       const _serviceSDLs = Object.values(serviceMap)
         .map((service) => service.schemaDefinition)
```

Reply:
Could you add that to the docs when writing them?

Reply:
Yes, I can :slightly_smiling_face: